actions.go

```diff
@@ -2,171 +2,176 @@ package neo4j
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
 	"github.com/cloudprivacylabs/opencypher/graph"
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
 )
 
-type createEdgeToSourceAndTarget struct {
-	Config
-	edge graph.Edge
+type neo4jAction interface {
+	Queue(neo4j.Transaction, *JobQueue) error
 }
 
-func (c createEdgeToSourceAndTarget) GetOCStmt(nodeIds map[graph.Node]int64) (string, map[string]interface{}) {
-	vars := make(map[string]interface{})
-	query := fmt.Sprintf("MATCH (f) WITH f MATCH (t) WHERE ID(f)=%d AND ID(t)=%d CREATE (f)-[%s %s]->(t)",
-		nodeIds[c.edge.GetFrom()],
-		nodeIds[c.edge.GetTo()],
-		c.MakeLabels([]string{c.edge.GetLabel()}),
-		c.MakeProperties(c.edge, vars))
-	return query, vars
+type JobQueue struct {
+	createNodes []graph.Node
+	createEdges []graph.Edge
+	deleteNodes []uint64
+	deleteEdges []uint64
 }
 
-func (c createEdgeToSourceAndTarget) Run(tx neo4j.Transaction, nodeIds map[graph.Node]int64) error {
-	query, vars := c.GetOCStmt(nodeIds)
-	_, err := tx.Run(query, vars)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-type createTargetFromSource struct {
+type DeleteEntity struct {
 	Config
-	edge graph.Edge
+	graph.Graph
+	entityId uint64
 }
 
-func (c createTargetFromSource) GetOCStmt(nodeIds map[graph.Node]int64) (string, map[string]interface{}) {
-	vars := make(map[string]interface{})
-	query := fmt.Sprintf("MATCH (from) WHERE ID(from) = %d CREATE (from)-[%s %s]->(to %s %s) RETURN to",
-		nodeIds[c.edge.GetFrom()],
-		c.MakeLabels([]string{c.edge.GetLabel()}),
-		c.MakeProperties(c.edge, vars),
-		c.MakeLabels(c.edge.GetTo().GetLabels().Slice()),
-		c.MakeProperties(c.edge.GetTo(), vars))
-	return query, vars
+type CreateEntity struct {
+	Config
+	graph.Graph
+	graph.Node
+	vars map[string]interface{}
 }
 
-func (c createTargetFromSource) Run(tx neo4j.Transaction, nodeIds map[graph.Node]int64) error {
-	query, vars := c.GetOCStmt(nodeIds)
-	idrec, err := tx.Run(query, vars)
-	if err != nil {
-		return err
+func (q *JobQueue) Run(tx neo4j.Transaction, cfg Config, nodeMap map[graph.Node]uint64, batchSize int) error {
+	const DEFAULT_BATCH_SIZE = 1000
+	if batchSize == 0 {
+		batchSize = DEFAULT_BATCH_SIZE
 	}
-	rec, err := idrec.Single()
-	if err != nil {
-		return err
+	vars := make(map[string]interface{})
+	for ix := 0; ix < len(q.deleteNodes); ix += batchSize {
```

Comment:
Instead of this, do:
``` 
for ;len(q.deleteNodes)>0; {
  batch=len(q.deleteNodes)
   if batch>batchSize {
      batch=batchSize
   }
  curBatch:=q.deleteNodes[:batch]
  q.deleteNodes=q.deleteNodes[batch:]
  ...
}
```